### DISCUSSION TITLE
How to flush pending operation buffer

### DISCUSSION THREAD

**Discussion by simon-abbott** [Automattic/mongoose#12412] [Q&A]
Hello,

I have a bit of a conundrum. I'm trying to drop a temporary database after a test run, but it's still there after the test completes. I have traced the issue, and I believe what is happening is that the database _is_ being dropped, but is then immediately being re-created by some buffered commands. Basically I think it's doing the following:

1. Connect
2. Run the tests
3. Drop the DB
4. Execute some commands that were buffered during the test run, which causes mongodb to re-create the database
5. Disconnect

What I would like is a way to wait until the command buffer is empty, so I can move step 4 before step 3, but there I can't find a `flush` function (or `wait`, or `pause`, or anything else that would signal when the queue is empty). Is there a way to do this cleanly? The only way I've found so far is to sleep for a while (`await new Promise((resolve) => setTimeout(resolve, 250));`) seems to do the trick), but I'd rather not wait for longer than necessary.

I'd also rather not turn off buffering entirely. I know that's an option, but I _do want_ buffering (plus our code is built assuming it's on, and I don't want to risk breaking anything), I just want to be able to flush the buffer manually.

**Comment by vkarpov15** [comment]
@simon-abbott there isn't currently a way to flush the buffer. However, you shouldn't need to flush the buffer after connecting - Mongoose only buffers commands before you initially connect (so before step 1).

Try enabling Mongoose debug mode, `mongoose.set('debug', true)`, to see what commands are executed in step (4).

Also, consider disabling `autoIndex` and `autoCreate` in your tests. `mongoose.set('autoIndex', false); mongoose.set('autoCreate', false);`. This will prevent Mongoose from automatically creating collections and indexes when you compile a model, which may help fix this problem.
